### DISCUSSION TITLE
How to use beta OCaml compiler via `nix shell`?

### DISCUSSION THREAD

**Discussion by Zeta611** [tweag/opam-nix#105] [Q&A] [answered]
Hi, I'd like to use the new `ocaml-base-compiler.5.3.0~rc1` via
```sh
𝝺 nix shell github:tweag/opam-nix#ocaml-base-compiler.5.3.0~rc1                                                                                                                at 11:08:45
error: flake 'github:tweag/opam-nix' does not provide attribute 'packages.aarch64-darwin.ocaml-base-compiler.5.3.0~rc1', 'legacyPackages.aarch64-darwin.ocaml-base-compiler.5.3.0~rc1' or 'ocaml-base-compiler.5.3.0~rc1'
```
but getting an error. It works with `.latest` tag though.

This maybe another question, but is there a way to use `utop` w/ `5.3.0~rc1` using `nix shell` as well?
Thanks in advance!

**Comment by balsoft** [comment, marked as answer]
You need to escape the attribute properly, like this:

```
nix shell 'github:tweag/opam-nix#ocaml-base-compiler."5.3.0~rc1"'
```

(mind the nested quotes)

**Reply by Zeta611** [reply]
Thanks a lot! But now I get that `5.3.0~rc1` is not provided:
```
error: flake 'github:tweag/opam-nix' does not provide attribute 'packages.x86_64-linux.ocaml-base-compiler."5.3.0~rc1"', 'legacyPackages.x86_64-linux.ocaml-base-compiler."5.3.0~rc1"' or 'ocaml-base-compiler."5.3.0~rc1"'
       Did you mean one of 5.0.0~rc1, 5.1.0~rc1, 5.2.0~rc1, 4.13.0~rc1 or 5.1.0~rc2?
```
The latest I can use is `5.3.0~beta1`, which is strange because I could use `5.3.0~beta2` in a flake+dune based project [here](https://github.com/tweag/opam-nix/issues/104).

**Reply by balsoft** [reply]
Ah, we probably need to update `opam-repository`.
